File: sample/src/main/java/com/github/dyvoker/throbber_sample/MainActivity.java
package com.github.dyvoker.throbber_sample;

import android.graphics.Color;
import android.os.Build;
import android.os.Bundle;
import android.view.ViewGroup;
import android.widget.FrameLayout;

import androidx.appcompat.app.AppCompatActivity;

import com.github.dyvoker.throbber.CircleDrawable;
import com.github.dyvoker.throbber.DensityConverter;
import com.github.dyvoker.throbber.ThrobberView;

public class MainActivity extends AppCompatActivity {

	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		setContentView(R.layout.activity_main);

		DensityConverter densityConverter = new DensityConverter(getResources().getDisplayMetrics());

		// Programmatically create throbber.
		ThrobberView throbberView = new ThrobberView(this);
		CircleDrawable circleDrawable = new CircleDrawable(
			Color.WHITE,
			densityConverter,
			0x80000000,
			2,
			0,
			2
		);
		if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN) {
			throbberView.setBackground(circleDrawable);
		} else {
			throbberView.setBackgroundDrawable(circleDrawable);
		}
		FrameLayout throbberContainer = findViewById(R.id.throbber_container);
		throbberContainer.addView(throbberView);
		ViewGroup.LayoutParams layoutParams = throbberView.getLayoutParams();
		layoutParams.width = (int) densityConverter.dpToPx(64);
		layoutParams.height = (int) densityConverter.dpToPx(64);
		throbberView.setLayoutParams(layoutParams);
	}
}

File: sample/src/main/java/com/github/dyvoker/throbber_sample/MainActivity.kt
package com.github.dyvoker.throbber_sample

import android.graphics.Color
import android.os.Build
import android.os.Bundle
import android.widget.FrameLayout
import androidx.appcompat.app.AppCompatActivity
import com.github.dyvoker.throbber.CircleDrawable
import com.github.dyvoker.throbber.DensityConverter
import com.github.dyvoker.throbber.ThrobberView

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val densityConverter = DensityConverter(resources.displayMetrics)
        // Programmatically create throbber.
        val throbberView = ThrobberView(this)
        val circleDrawable = CircleDrawable(
                Color.WHITE,
                densityConverter,
                -0x80000000,
                2f,
                0f,
                2f
        )
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN) {
            throbberView.background = circleDrawable
        } else {
            throbberView.setBackgroundDrawable(circleDrawable)
        }
        val throbberContainer = findViewById<FrameLayout>(R.id.throbber_container)
        throbberContainer.addView(throbberView)
        val layoutParams = throbberView.layoutParams
        layoutParams.width = densityConverter.dpToPx(64f).toInt()
        layoutParams.height = densityConverter.dpToPx(64f).toInt()
        throbberView.layoutParams = layoutParams
    }
}